### ISSUE TITLE
Misleading error message if missing checkers.

### ISSUE THREAD

**Issue by castelao** [ioos/compliance-checker#306]
If running ComplianceChecker.run_checker() without:

check_suite = CheckSuite()
check_suite.load_all_available_checkers()

would return a message:

```
compliance-checker/compliance_checker/runner.pyc in stdout_output(cos,
cs, score_groups, verbose, limit)
     96             else:
     97                 cs.verbose_output_generation(groups, limit,
points, out_of)
---> 98         return groups
     99
    100     @classmethod

UnboundLocalError: local variable 'groups' referenced before assignment
```

Looks like that the lack of any checker is the actual problem, so wouldn't be better to
fail earlier and raise that? Is there any reason to check without loading any checker?

**Comment by benjwadams** [comment]
@castelao, do you have a working commit for this?  It looks like the commit referenced fixes things, but I'd prefer an if statement over assert in non testing code due to Python's -O switch, which can skip asserts.

**Comment by lukecampbell** [comment]
I think we added an assert somewhere.

**Comment by benjwadams** [comment]
Fixed by #416, closing.

**Comment by castelao** [comment]
@benjwadams and @lukecampbell Sorry for haven't responded that on time!
